### ISSUE TITLE
set_collision_layer_bit and set_collision_mask_bit  create an illusion in the inspector, but they do not affect physics.

### ISSUE THREAD

**Issue by OlexiyKravchuk** [godotengine/godot#29564]
Godot_v3.1.1-stable_win64.exe
Windows 7

Switching collision layers and masks for physical bodies using the set_collision_layer_bit (№, true / false) and set_collision_mask_bit (№, true / false) functions in the script has no effect on the collisions in the game, but it can be tracked in the inspector that the switching going fine. In addition, if you manually switch layers in the Inspector before starting the game, they will remain in this position both after the game is started and in the script when the functions are called. In other words, the inspector says what works, but in fact not.

**Comment by KoBeWi** [comment]
Which bodies exactly? Any reproduction project?

**Comment by avencherus** [comment]
I have not been experiencing this in my physics work lately, I'd also like to see how to reproduce it.

**Comment by OlexiyKravchuk** [comment]
@KoBeWi @avencherus
Forgive me, it was probably a false alarm, I could not find out what the reason was and why physics refused to work properly.
I have deadlines, and I have been spending on this problem for 3 days already and I can no longer look for reasons, so I just redid everything in a new way, and now everything seems to be working.
I now have only one guess, why it did not work in the first version ...
To realize the explosions that caused damage to the area at a defined point in time, I used the method ($"Explosion/CollisionShape2D".set_deferred("disabled", false)) to include a detection zone, which was to take all the bodies in the radius of the lesion and call into them the function of dealing damage.
Now I did everything the other way, avoiding this function.
My guess was that the physics server did not have time to process the overlap of bodies in this frame (in which the function included the form), and waited for the next frame, but then it would be too late to process the event, since by the time of the next frame the object should be destroyed both from one and from the other side of the interaction.
Based on this, I close this problem.
